In [70]:
import numpy as np
import pandas as pd

In [71]:
fallecidos=pd.read_csv('https://raw.githubusercontent.com/MinCiencia/Datos-COVID19/master/output/producto50/DefuncionesDEIS_confirmadosPorComuna_std.csv')
vc1= pd.read_csv('https://raw.githubusercontent.com/MinCiencia/Datos-COVID19/master/output/producto80/vacunacion_comuna_1eraDosis_std.csv')
riq= pd.read_csv('https://raw.githubusercontent.com/leoferres/gitiele356b/main/data/hdi.csv')

In [72]:
fallecidos=fallecidos[fallecidos.Region=='Metropolitana']
fallecidos=fallecidos[fallecidos.Comuna!='Total']
fallecidos['Fecha'] = pd.to_datetime(fallecidos['Fecha'], dayfirst=True)
fallecidos=fallecidos.sort_values(['Fecha'])
fallecidos=fallecidos.reset_index()
fallecidos=fallecidos.drop(['index'], axis=1 )
fallecidos=fallecidos.drop(['Codigo region'], axis=1 )


In [73]:
fallecidos=fallecidos.drop(range(0,14999), axis=0)
fallecidos=fallecidos.reset_index()
fallecidos=fallecidos.drop(['index'], axis=1 )
fallecidos


Region                     Comuna  Codigo comuna  Poblacion  \
0      Metropolitana                Puente Alto        13201.0   645909.0   
1      Metropolitana                   Pudahuel        13124.0   253139.0   
2      Metropolitana                Providencia        13123.0   157749.0   
3      Metropolitana                     Pirque        13202.0    30433.0   
4      Metropolitana                      Nunoa        13120.0   250192.0   
...              ...                        ...            ...        ...   
10383  Metropolitana               Lo Barnechea        13115.0   124076.0   
10384  Metropolitana                  Lo Espejo        13116.0   103865.0   
10385  Metropolitana                   Lo Prado        13117.0   104403.0   
10386  Metropolitana                 Huechuraba        13107.0   112528.0   
10387  Metropolitana  Desconocido Metropolitana            NaN        NaN   

           Fecha  Defunciones  
0     2020-12-24          4.0  
1     2020-12-24          0.0  
2     2020-12-24          2.0  
3     2020-12-24          0.0  
4     2020-12-24          0.0  
...          ...          ...  
10383 2021-07-07          0.0  
10384 2021-07-07          0.0  
10385 2021-07-07          0.0  
10386 2021-07-07          0.0  
10387 2021-07-07          NaN  

[10388 rows x 6 columns]

In [74]:
fallecidos=fallecidos.drop([10387], axis=0 )
fallecidos

Region        Comuna  Codigo comuna  Poblacion      Fecha  \
0      Metropolitana   Puente Alto        13201.0   645909.0 2020-12-24   
1      Metropolitana      Pudahuel        13124.0   253139.0 2020-12-24   
2      Metropolitana   Providencia        13123.0   157749.0 2020-12-24   
3      Metropolitana        Pirque        13202.0    30433.0 2020-12-24   
4      Metropolitana         Nunoa        13120.0   250192.0 2020-12-24   
...              ...           ...            ...        ...        ...   
10382  Metropolitana    Las Condes        13114.0   330759.0 2021-07-07   
10383  Metropolitana  Lo Barnechea        13115.0   124076.0 2021-07-07   
10384  Metropolitana     Lo Espejo        13116.0   103865.0 2021-07-07   
10385  Metropolitana      Lo Prado        13117.0   104403.0 2021-07-07   
10386  Metropolitana    Huechuraba        13107.0   112528.0 2021-07-07   

       Defunciones  
0              4.0  
1              0.0  
2              2.0  
3              0.0  
4              0.0  
...            ...  
10382          0.0  
10383          0.0  
10384          0.0  
10385          0.0  
10386          0.0  

[10387 rows x 6 columns]

In [75]:
fallecidos2=fallecidos.groupby(['Comuna','Codigo comuna']).sum().reset_index()
fallecidos2=fallecidos2.drop(['Poblacion'], axis=1 )
fallecidos2

Comuna  Codigo comuna  Defunciones
0                 Alhue        13502.0          3.0
1                  Buin        13402.0         68.0
2       Calera de Tango        13403.0         23.0
3             Cerrillos        13102.0         68.0
4           Cerro Navia        13103.0        221.0
5                Colina        13301.0         89.0
6              Conchali        13104.0        165.0
7              Curacavi        13503.0         22.0
8             El Bosque        13105.0        191.0
9              El Monte        13602.0         32.0
10     Estacion Central        13106.0        119.0
11           Huechuraba        13107.0        114.0
12        Independencia        13108.0        114.0
13        Isla de Maipo        13603.0         21.0
14          La Cisterna        13109.0        120.0
15           La Florida        13110.0        407.0
16            La Granja        13111.0        151.0
17           La Pintana        13112.0        180.0
18             La Reina        13113.0         75.0
19                Lampa        13302.0        103.0
20           Las Condes        13114.0        195.0
21         Lo Barnechea        13115.0         61.0
22            Lo Espejo        13116.0        143.0
23             Lo Prado        13117.0        117.0
24                Macul        13118.0        120.0
25                Maipu        13119.0        351.0
26          Maria Pinto        13504.0          6.0
27            Melipilla        13501.0        122.0
28                Nunoa        13120.0        158.0
29        Padre Hurtado        13604.0         47.0
30                Paine        13404.0         34.0
31  Pedro Aguirre Cerda        13121.0        138.0
32             Penaflor        13605.0         73.0
33            Penalolen        13122.0        259.0
34               Pirque        13202.0         30.0
35          Providencia        13123.0        145.0
36             Pudahuel        13124.0        180.0
37          Puente Alto        13201.0        600.0
38            Quilicura        13125.0        132.0
39        Quinta Normal        13126.0        128.0
40             Recoleta        13127.0        165.0
41                Renca        13128.0        166.0
42         San Bernardo        13401.0        338.0
43          San Joaquin        13129.0        108.0
44    San Jose de Maipo        13203.0         22.0
45           San Miguel        13130.0        148.0
46            San Pedro        13505.0          8.0
47            San Ramon        13131.0        157.0
48             Santiago        13101.0        223.0
49            Talagante        13601.0         76.0
50               Tiltil        13303.0         18.0
51             Vitacura        13132.0         38.0

In [76]:
vc1['Fecha']=pd.to_datetime(vc1['Fecha'], dayfirst=True)
vc1=vc1[vc1.Region=='Metropolitana']
vc1=vc1.reset_index()
vc1=vc1.drop(['index'], axis=1 )
vc1=vc1.drop(['Codigo region'], axis=1 )
vc1=vc1.drop([10599], axis=0 )
vc1=vc1.rename(columns={'Primera Dosis':'Primera'})
vc1

Region           Comuna  Codigo comuna      Fecha  Primera
0      Metropolitana            Alhue        13502.0 2020-12-24      0.0
1      Metropolitana             Buin        13402.0 2020-12-24      0.0
2      Metropolitana  Calera de Tango        13403.0 2020-12-24      0.0
3      Metropolitana        Cerrillos        13102.0 2020-12-24      0.0
4      Metropolitana      Cerro Navia        13103.0 2020-12-24      0.0
...              ...              ...            ...        ...      ...
10594  Metropolitana        San Ramon        13131.0 2021-07-11    196.0
10595  Metropolitana         Santiago        13101.0 2021-07-11      0.0
10596  Metropolitana        Talagante        13601.0 2021-07-11      0.0
10597  Metropolitana           Tiltil        13303.0 2021-07-11      0.0
10598  Metropolitana         Vitacura        13132.0 2021-07-11      0.0

[10599 rows x 5 columns]

In [77]:
vc1=vc1.groupby(['Comuna','Codigo comuna']).sum().reset_index()
vc1

Comuna  Codigo comuna   Primera
0                 Alhue        13502.0    6232.0
1                  Buin        13402.0   74943.0
2       Calera de Tango        13403.0   21869.0
3             Cerrillos        13102.0   50012.0
4           Cerro Navia        13103.0   74917.0
5                Colina        13301.0   94464.0
6              Conchali        13104.0   99163.0
7              Curacavi        13503.0   24079.0
8             El Bosque        13105.0  101296.0
9              El Monte        13602.0   23088.0
10     Estacion Central        13106.0   97504.0
11           Huechuraba        13107.0   82033.0
12        Independencia        13108.0   93925.0
13        Isla de Maipo        13603.0   26950.0
14          La Cisterna        13109.0   60910.0
15           La Florida        13110.0  283353.0
16            La Granja        13111.0   87916.0
17           La Pintana        13112.0   86721.0
18             La Reina        13113.0  118911.0
19                Lampa        13302.0   62202.0
20           Las Condes        13114.0  290489.0
21         Lo Barnechea        13115.0  104054.0
22            Lo Espejo        13116.0   60930.0
23             Lo Prado        13117.0   77929.0
24                Macul        13118.0   89311.0
25                Maipu        13119.0  308034.0
26          Maria Pinto        13504.0   10179.0
27            Melipilla        13501.0   92680.0
28                Nunoa        13120.0  196017.0
29        Padre Hurtado        13604.0   52792.0
30                Paine        13404.0   55131.0
31  Pedro Aguirre Cerda        13121.0   62005.0
32             Penaflor        13605.0   60669.0
33            Penalolen        13122.0  147589.0
34               Pirque        13202.0   22160.0
35          Providencia        13123.0  207998.0
36             Pudahuel        13124.0  146022.0
37          Puente Alto        13201.0  299428.0
38            Quilicura        13125.0  125223.0
39        Quinta Normal        13126.0   79399.0
40             Recoleta        13127.0  107345.0
41                Renca        13128.0   84419.0
42         San Bernardo        13401.0  169040.0
43          San Joaquin        13129.0   70761.0
44    San Jose de Maipo        13203.0   15052.0
45           San Miguel        13130.0   88954.0
46            San Pedro        13505.0    8097.0
47            San Ramon        13131.0   54712.0
48             Santiago        13101.0  289664.0
49            Talagante        13601.0   52017.0
50               Tiltil        13303.0   13898.0
51             Vitacura        13132.0  104057.0

In [78]:
df_def=pd.merge(fallecidos2, vc1, on=['Comuna','Codigo comuna'], how='left')

In [79]:
riq=riq.rename(columns={'comuna':'Codigo comuna'})
riq=riq.loc[:,['Codigo comuna','HDI']]


In [80]:
df_def=pd.merge(df_def, riq, on=['Codigo comuna'], how='left')

In [81]:
df_def

Comuna  Codigo comuna  Defunciones   Primera       HDI
0                 Alhue        13502.0          3.0    6232.0  0.742609
1                  Buin        13402.0         68.0   74943.0  0.766439
2       Calera de Tango        13403.0         23.0   21869.0  0.715509
3             Cerrillos        13102.0         68.0   50012.0  0.764616
4           Cerro Navia        13103.0        221.0   74917.0  0.718874
5                Colina        13301.0         89.0   94464.0  0.690036
6              Conchali        13104.0        165.0   99163.0  0.749584
7              Curacavi        13503.0         22.0   24079.0  0.738340
8             El Bosque        13105.0        191.0  101296.0  0.742015
9              El Monte        13602.0         32.0   23088.0  0.728127
10     Estacion Central        13106.0        119.0   97504.0  0.767604
11           Huechuraba        13107.0        114.0   82033.0  0.712375
12        Independencia        13108.0        114.0   93925.0  0.763935
13        Isla de Maipo        13603.0         21.0   26950.0  0.776123
14          La Cisterna        13109.0        120.0   60910.0  0.850780
15           La Florida        13110.0        407.0  283353.0  0.795888
16            La Granja        13111.0        151.0   87916.0  0.726793
17           La Pintana        13112.0        180.0   86721.0  0.689951
18             La Reina        13113.0         75.0  118911.0  0.935337
19                Lampa        13302.0        103.0   62202.0  0.718193
20           Las Condes        13114.0        195.0  290489.0  0.955683
21         Lo Barnechea        13115.0         61.0  104054.0  0.854046
22            Lo Espejo        13116.0        143.0   60930.0  0.743558
23             Lo Prado        13117.0        117.0   77929.0  0.752665
24                Macul        13118.0        120.0   89311.0  0.838293
25                Maipu        13119.0        351.0  308034.0  0.799755
26          Maria Pinto        13504.0          6.0   10179.0  0.723305
27            Melipilla        13501.0        122.0   92680.0  0.720423
28                Nunoa        13120.0        158.0  196017.0  0.946392
29        Padre Hurtado        13604.0         47.0   52792.0  0.719437
30                Paine        13404.0         34.0   55131.0  0.721614
31  Pedro Aguirre Cerda        13121.0        138.0   62005.0  0.771589
32             Penaflor        13605.0         73.0   60669.0  0.760697
33            Penalolen        13122.0        259.0  147589.0  0.731645
34               Pirque        13202.0         30.0   22160.0  0.776297
35          Providencia        13123.0        145.0  207998.0  0.968496
36             Pudahuel        13124.0        180.0  146022.0  0.731874
37          Puente Alto        13201.0        600.0  299428.0  0.740045
38            Quilicura        13125.0        132.0  125223.0  0.753152
39        Quinta Normal        13126.0        128.0   79399.0  0.762931
40             Recoleta        13127.0        165.0  107345.0  0.728831
41                Renca        13128.0        166.0   84419.0  0.726315
42         San Bernardo        13401.0        338.0  169040.0  0.726599
43          San Joaquin        13129.0        108.0   70761.0  0.791456
44    San Jose de Maipo        13203.0         22.0   15052.0  0.753607
45           San Miguel        13130.0        148.0   88954.0  0.840925
46            San Pedro        13505.0          8.0    8097.0  0.679979
47            San Ramon        13131.0        157.0   54712.0  0.710381
48             Santiago        13101.0        223.0  289664.0  0.861222
49            Talagante        13601.0         76.0   52017.0  0.759548
50               Tiltil        13303.0         18.0   13898.0  0.735826
51             Vitacura        13132.0         38.0  104057.0  0.994584

In [82]:
import statsmodels.formula.api as sfm

In [92]:
model=sfm.ols(formula='Defunciones ~ HDI+Primera',data=df_def)
fit = model.fit()

In [93]:
fit.params

Intercept    470.460794
HDI         -618.264405
Primera        0.001401
dtype: float64

In [94]:
print(fit.summary())

                            OLS Regression Results                            
Dep. Variable:            Defunciones   R-squared:                       0.789
Model:                            OLS   Adj. R-squared:                  0.780
Method:                 Least Squares   F-statistic:                     91.53
Date:                Mon, 12 Jul 2021   Prob (F-statistic):           2.84e-17
Time:                        22:23:52   Log-Likelihood:                -277.31
No. Observations:                  52   AIC:                             560.6
Df Residuals:                      49   BIC:                             566.5
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    470.4608     80.599      5.837      0.0

De acuerdo al R2, el 78.9% de la variacion de las defunciones por covid son explicadas por las variables de la cantidad de primeras dosis aplicadas y el HDI por comuna

El intercept, indica que si el HDI y el numero de primeras dosis son cero, las muertes serian de 470 personas

En cuanto a los factores podemos observar que a mayor riqueza(HDI) menor son las muertes y que de manera extraña aunque con baja influencia a mayor cantidad de primeras dosis mayor es la cantidad de muertes

Cabe notar que los resultados nos arrojan que puede existir una fuerte multicolinealidad entre las variables HDI y N° de primeras dosis 

Una variable que se podria agregar al modelo es la cantidad de segundas dosis aplicadas ya que de manera intuitiva la segunda dosis ofrece mayor proteccion y deberia hacer que el modelo se acerque un poco mas a la realidad actual que se esta viviendo.